### ISSUE TITLE
Default generated endpoints places variables into the root object: should variables be nested into a property?

### ISSUE THREAD

**Issue by StephenOTT** [kiegroup/kogito-runtimes#1892]
### Description

When you start a instance, you are submitted variables.  But when you GET an instance you are receiving process instance ~metadata and process variables (or task/local variables).

The variables are mixed into the name level as the metadata.  Should this data be moved into a nested property?  such as "variables", or "data" (data because that is the name used internally).

I suggest this because as processes get complex, the number of variables increases and it is quickly easily to get mixed up on what are variables and what are process/task metadata.  Then scale that out when you start to add new properties in the future, and could easily have variable name conflicts with newly added metadata. 

Example: imagine one day you added a pre-calculated `duration` property for instances that showed the total duration (without the need to calculate it).  This could easily become a conflict with business variables.

**Comment by fjtirado** [comment]
@cristianonicolai @evacchi @ricardozanini I agree with @StephenOTT rationale. Although currently we only have id as metadata, when we add business key, thing might start getting messy. I think wrapping model under a model field is a must. 
I do not see as clear the need to wrap metadata under a different objet, because metadata fields the same one for all processes and there is not risk for name collision, which is the main argument to do that from my point of view

**Comment by StephenOTT** [comment]
Anything that is a property name created by the modeler could cause collision. so should be isolated.

**Comment by cristianonicolai** [comment]
the main idea behind keeping it at root level is that we avoid having Kogito specific structure in the response, we want to focus on the domain and so making it extremely easy to be consumed/integrated elsewhere. In saying that I agree that at some point things can get messy. Maybe we could have strategies to deal with it so people can decide how to best approach this based on their use case. For instance, we could also have Kogito specific details sent as headers instead.

**Comment by StephenOTT** [comment]
Having the "data" in the body makes it easy for consumption.  Getting into headers starts to create complexity for regular business data consumption.  
What about an optional data wrapper that could be set in config? 

@ChristianLowe if you do not have a model/data wrapper, how do you intend to deal with property name collision?  Imagine you decided in v1.20 to start returning a `status` property, `dueDate` or an `assignee` property on user tasks: these could easily be process variables already in use by modellers.

**Comment by ricardozanini** [comment]
I think it's doable to have this feature configurable. This way, the consumer will understand the caveats and implications to their architecture. By default, we can add this information to the headers as we do for CloudEvents and OpenAPI client calls.

**Comment by fjtirado** [comment]
In my opinion, If we want to keep model data at root level, then we need to move the not model info under a different object. I propose to call that field "_meta" to make name colision really unlikely. 
This idea is taken from HATEOAS impl named [HAL](https://en.wikipedia.org/wiki/Hypertext_Application_Language), which use _links for additional info.
